src/Data/Connection/PostObjectConnectionResolver.php

```diff
@@ -322,20 +327,22 @@ public function get_query_args() {
 					true
 				) ) {
 					$query_args['orderby'] = esc_sql( $orderby_input['field'] );
-				} elseif ( ! empty( $orderby_input['field'] ) ) {
 
-					$order = $orderby_input['order'];
+					// If we're ordering explicitly, there's no reason to check other orderby inputs.
+					break;
```

Comment:
The point of this conditional is that `post__in` etc overwrites all other sorting criteria. As such, we should bail early instead of overwriting it if the user unwittingly orders by this _and_ other fields.